fanficfare/adapters/adapter_spiritfanfictioncom.py

```diff
@@ -0,0 +1,350 @@
+                i = parse_until_br('kudos', i+1, info_contents)
+            elif content.text.strip() == 'Comentários':
+                i = parse_until_br('comments', i+1, info_contents)
+            elif content.text.strip() == 'Listas de leitura':
+                i = parse_until_br('bookmarks', i+1, info_contents)
+            elif content.text.strip() == 'Palavras':
+                i = parse_until_br('numWords', i+1, info_contents)
+            elif content.text.strip() == 'Concluído':
+                i = parse_until_br('status', i+1, info_contents)
+            elif content.text.strip() == 'Categorias':
+                i = parse_until_br('category', i+1, info_contents)
+            elif content.text.strip() == 'Personagens':
+                i = parse_until_br('characters', i+1, info_contents)
+            elif content.text.strip() == 'Tags':
+                i = parse_until_br('freeformtags', i+1, info_contents)
+            else:
+                i += 1
+
+        # Classificação, Gêneros e Avisos
+        # Finding div element with class "clearfix baixo"
+        div_element = soup.find('div', {'class': 'clearfix baixo'})
+
+        # Finding div element with class "classificacao"
+        classificacao_element = div_element.find('div', class_='classificacao')
+
+        # Extracting last word from class name
+        if classificacao_element and 'class' in classificacao_element.attrs:
+            class_value = classificacao_element.attrs['class']
+            rating = class_value[-1].split('-')[-1]
+
+        # Extracting text content "Gêneros" and "Avisos"
+        contents = classificacao_element.find_next('div').contents
+        for i, content in enumerate(contents):
+            if content.text.strip() == 'Gêneros:':
+                parse_until_br('genre', i+1, contents)
+            if content.text.strip() == 'Avisos:':
+                parse_until_br('warnings', i+1, contents)
+
+    ## Normalize chapter URLs in case of title change
+    def normalize_chapterurl(self,url):
+        #https://www.spiritfanfiction.com/historia/story-name-1234/capitulo56
+        url = re.sub(r"https?://("+self.getSiteDomain()+r"/historia/\d+/capitulo\d+)$",
+                     r"https://\1",url)
+        return url
+
+
+    def getChapterText(self, url):
+        logger.debug('Getting chapter text from: %s' % url)
+
+        save_chapter_soup = self.make_soup(self.get_request(url))
+        ## need save_chapter_soup for .new_tag()
+        save_chapter = BeautifulSoup("", "html.parser").new_tag("div")
+
+        chapter_dl_soup = self.make_soup(self.get_request(url))
+        if None == chapter_dl_soup:
+            raise exceptions.FailedToDownload("Error downloading Chapter: %s!  Missing required element!" % url)
+        div_chapter = chapter_dl_soup.find('div', {'class':'clearfix'})
+        chapter_text = div_chapter.find('div', class_='texto-capitulo')
+
+        exclude_notes=self.getConfigList('exclude_notes')
```

Comment:
You need to add a `[www.spiritfanfiction.com]` section to [defaults.ini](https://github.com/JimmXinu/FanFicFare/blob/main/fanficfare/defaults.ini) and [plugin-defaults.ini](https://github.com/JimmXinu/FanFicFare/blob/main/calibre-plugin/plugin-defaults.ini) to document the adapter specific settings, like `exclude_notes`.

Also all site specific `extra_valid_entries`, you have several above, like `hits`, `bookmarks`, etc.